packages/x-data-grid/src/components/GridRow.tsx

```diff
@@ -477,8 +451,7 @@ const GridRow = React.forwardRef<HTMLDivElement, GridRowProps>(function GridRow(
         style={{ width: offsetLeft }}
       />
       {cells}
-      {emptyCellWidth > 0 && <EmptyCell width={emptyCellWidth} />}
-      {rightCells.length > 0 && <div role="presentation" className={gridClasses.filler} />}
+      <div role="presentation" className={clsx(gridClasses.cell, gridClasses.cellEmpty)} />
```

Comment:
Rendering the empty cell all the time improves resizing of the right-pinned cells

**Before**

https://github.com/user-attachments/assets/779b759b-4cd7-4948-9d4c-2fc74a8f7b41

**After**

https://github.com/user-attachments/assets/7546bd15-9830-4217-973d-395a31c0220c

Reply:
Ideally the actions column would also move with the right pinned column in the after...